demo-app/src/tx_verifier.rs

```diff
@@ -0,0 +1,99 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use sov_modules_api::{Context, Signature, Spec};
+use sovereign_sdk::jmt::SimpleHasher;
+use sovereign_sdk::serial::Decode;
+use std::{io::Cursor, marker::PhantomData};
+
+/// RawTx represents a serialized rollup transaction received from the DA.
+#[derive(Debug, PartialEq, BorshDeserialize, BorshSerialize)]
+pub struct RawTx {
+    pub(crate) data: Vec<u8>,
+}
+
+/// Transaction represents a deserialized RawTx.
+#[derive(Debug, PartialEq, Eq, Clone, BorshDeserialize, BorshSerialize)]
+pub struct Transaction<C: sov_modules_api::Context> {
+    pub signature: C::Signature,
+    pub pub_key: C::PublicKey,
+    pub runtime_msg: Vec<u8>,
+    pub nonce: u64,
+}
+
+/// VerifiedTx is a Transaction after verification.
+pub(crate) struct VerifiedTx<C: Context> {
+    pub(crate) sender: C::PublicKey,
+    pub(crate) runtime_msg: Vec<u8>,
+    pub(crate) _nonce: u64,
+}
+
+/// TxVerifier encapsulates Transaction verification.
+pub(crate) trait TxVerifier {
+    type Context: Context;
+
+    /// Runs stateless checks against RawTxs.
+    fn verify_txs_stateless(
```

Comment:
Should add a version of this method (`verify_tx_stateless`) that works on a single tx? I think that will be useful for sequencers. We can keep the batch method too if we want, and just add a default impl that calls `verify_tx_stateless` in a loop

Reply:
done